feat: add C implementation for `stats/base/dists/levy/quantile`

saurabhraghuvanshii opened the pull request:
Resolves #3730 .

## Description

> What is the purpose of this pull request?

This pull request:

-  add c implementation for `@stdlib/stats/base/dists/levy/quantile`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3730 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
feat: add c implementation for stats/base/dists/levy/quantile

saurabhraghuvanshii committed:
chore: update copyright year of include.gypi and fix lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

saurabhraghuvanshii commented:
/stdlib help

saurabhraghuvanshii commented:
/stdlib help

saurabhraghuvanshii commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -31,18 +32,26 @@ var quantile = require( './../lib' );
+	mu = new Float64Array( len );
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		p[ i ] = randu();
```
```suggestion
		p[ i ] = unifrom( 0.0, 1.0 );
```
as the logic in `benchmark.c` should resonate with the js file so for added clarity use `uniform` function to generate the parameters through out the js file.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -31,18 +32,26 @@ var quantile = require( './../lib' );
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		p[ i ] = randu();
+		mu[ i ] = ( randu() * 20.0 ) - 10.0;
```
```suggestion
		mu[ i ] = uniform( -10.0, 10.0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -31,18 +32,26 @@ var quantile = require( './../lib' );
+	for ( i = 0; i < len; i++ ) {
+		p[ i ] = randu();
+		mu[ i ] = ( randu() * 20.0 ) - 10.0;
+		c[ i ] = ( randu() * 5.0 ) + EPS;
```
```suggestion
		c[ i ] = uniform( EPS, 5.0 );
```

Neerajpathak07 replied:
You should also add the similar logic for factory code block as well

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+
+	for ( i = 0; i < 100; i++ ) {
+		p[ i ] = random_uniform( 0.0, 100.0 );
+		mu[ i ] = random_uniform( 0.0, 100.0 ) - 50.0;
```
```suggestion
		mu[ i ] = random_uniform( -50.0, 50.0 ) ;
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	for ( i = 0; i < 100; i++ ) {
+		p[ i ] = random_uniform( 0.0, 100.0 );
+		mu[ i ] = random_uniform( 0.0, 100.0 ) - 50.0;
+		c[ i ] = random_uniform( 0.0, 20.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
```
```suggestion
		c[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+
+	for ( i = 0; i < 25; i++ ) {
+		p = random_uniform( 0.0, 10.0 );
+		mu = random_uniform( 0.0, 10.0 ) - 5.0;
```
```suggestion
		mu = random_uniform( -5.0, 5.0 ) ;
```

saurabhraghuvanshii replied:
It is new convetion only! , wait I will try

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/src/main.c:
```diff
@@ -0,0 +1,49 @@
+* @param p	 input value
+* @param mu	location parameter
+* @param c	 scale parameter
+* @return	  evaluated quantile function
```
also would recommend to stack these on top of one another.

saurabhraghuvanshii replied:
ok lets see

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
```diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_levy_quantile( p, mu, c )
+
+Returns the [quantile function][quantile-function] for a [Lévy][levy-distribution] distribution with parameters `mu` (location parameter) and `c` (scale parameter) at a probability `p`.
```
```suggestion
Returns the quantile function for a Lévy distribution with location parameter `mu`  and scale parameter `c`.
```
should match the JSDoc API call mentioned above.
SInce you might have referenced it in `main.c` and `quantile.h` you'll have to change it there as well.

saurabhraghuvanshii replied:
thanks for review . I am doing any more changes if needed then tell me.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,74 @@
+	p = new Float64Array( len );
+	mu = new Float64Array( len );
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
```
make the same changes for `native` files as well.

saurabhraghuvanshii replied:
ok , I have a problem ,  I always confuse how to update readme  what should add .

Neerajpathak07 replied:
alright so in the `README.md` file what we do is basically put the necessary C example code in triple backticks to when you view it on github you get the idea of it's implementation.
For this instance you never add benchmark code into markdown file as we only want to showcase the necessary JSDOC api's , javascript examples and C examples.
With me so far??

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		p = random_uniform( 0.0, 10.0 );
```
also I just referred `examples/index.js`  where in the values of `p` ,`mu` and `c` do not correspond with your C file.

Neerajpathak07 replied:
```suggestion
		p = random_uniform( 0.0, 1.0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+
+	for ( i = 0; i < 25; i++ ) {
+		p = random_uniform( 0.0, 10.0 );
+		mu = random_uniform( 0.0, 10.0 ) - 5.0;
```
```suggestion
		mu = random_uniform( 0.0, 10.0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+	for ( i = 0; i < 25; i++ ) {
+		p = random_uniform( 0.0, 10.0 );
+		mu = random_uniform( 0.0, 10.0 ) - 5.0;
+		c = random_uniform( 0.0, 20.0 );
```
```suggestion
		c = random_uniform( 0.0, 10.0 );
```

Neerajpathak07 replied:
these should be the new values for `p` , `mu` and `c`

saurabhraghuvanshii replied:
done .

saurabhraghuvanshii replied:
review new commit if any changes let me know

saurabhraghuvanshii committed:
refactor: updates file with uniform and values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/include/stdlib/stats/base/dists/levy/quantile.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Returns the quantile for a Lévy distribution with location `mu` and scale `c` at a probability `p`.
```
Just make sure the this api call is the one similar to the JSDOC api in markdown also under `stats_base_dists_levy_quantile`

Neerajpathak07 replied:
you can refer to `main.c` for the same

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
```diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+    for ( i = 0; i < 25; i++ ) {
+        p = random_uniform();
+        mu = random_uniform( 0.0, 10.0 ) - 5.0;
+        c = random_uniform( 0.0, 20.0 );
```
add the updated `example.c` file in the markdown file,

saurabhraghuvanshii replied:
done

saurabhraghuvanshii committed:
chore: update value in Redame and some in quantile.h

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		p[ i ] = random_uniform( 0.0, 100.0 );
```
```suggestion
		p[ i ] = random_uniform( 0.0, 1.0 );
```

saurabhraghuvanshii replied:
how it changes to 100

Neerajpathak07 replied:
refer the corresponding `benchmark.js` file

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+
+	for ( i = 0; i < 100; i++ ) {
+		p[ i ] = random_uniform( 0.0, 100.0 );
+		mu[ i ] = random_uniform( -50.0, 50.0 );
```
```suggestion
		mu[ i ] = random_uniform( -10.0, 10.0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	for ( i = 0; i < 100; i++ ) {
+		p[ i ] = random_uniform( 0.0, 100.0 );
+		mu[ i ] = random_uniform( -50.0, 50.0 );
+		c[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 );
```
```suggestion
		c[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 5.0 );
```

saurabhraghuvanshii replied:
can i ask how do you know where what values should be puts . I always confuse what value should be put I referred file but did'nt come with idea.

Neerajpathak07 replied:
it's not that difficult, so while writing  C code for `benchmark` or `example` start by referring the corresponding js file and see which values are assigned to the parameters the function call and around which they are wrapped can be different.

saurabhraghuvanshii replied:
i do same thing even i see some other Pr reference but for some file values goes half it confuse

Neerajpathak07 replied:
were these the merged PR's you were referring to ?

saurabhraghuvanshii replied:
yeah I always refer merged pr but on some C implementation Prs are not merged so i refer ayush pr mostly.

Neerajpathak07 replied:
oh ok

Neerajpathak07 replied:
> it's not that difficult, so while writing C code for `benchmark` or `example` start by referring the corresponding js file and see which values are assigned to the parameters the function call and around which they are wrapped can be different.

But this should be the more prominent approach.

saurabhraghuvanshii replied:
I will follow all possible way. thanks for help.

saurabhraghuvanshii replied:
one more help can you do , when i try to push it after updating . commit messup with show many commits, what is best way to sync working branch from upstream and to push the changes.

Neerajpathak07 replied:
when I face something similar what works for me is I fetch the commits from origin of the same branch itself which then undergoes a merge branch commit.
In your case you could try :- `git pull origin levy-quantile`
This should pull all the commits from the remote branch

saurabhraghuvanshii replied:
done

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into levy-quantile

saurabhraghuvanshii committed:
chore: update benchmark.c

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/manifest.json:
```diff
@@ -0,0 +1,89 @@
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/napi/ternary",
+          "@stdlib/math/base/assert/is-nan",
+          "@stdlib/constants/float64/pinf",
+          "@stdlib/constants/float64/ninf",
+          "@stdlib/math/base/special/erfcinv",
+          "@stdlib/constants/float64/eps"
```
```suggestion
          "@stdlib/math/base/napi/ternary",
          "@stdlib/math/base/assert/is-nan",
          "@stdlib/math/base/special/erfcinv"
```
you need to remove the header files which are not in use.

Neerajpathak07 replied:
do the same below as well.

saurabhraghuvanshii replied:
builds failed there fore i added all this .

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/manifest.json:
```diff
@@ -0,0 +1,89 @@
+          "@stdlib/constants/float64/ninf",
+          "@stdlib/constants/float64/pinf",
+          "@stdlib/constants/float64/eps",
+          "@stdlib/random/base/uniform"
```
```suggestion

```
no need to add uniform header file

saurabhraghuvanshii replied:
but without uniform builds are failed, I tried , there fore added it

Neerajpathak07 replied:
apply the suggested changes and try once more.
Or if you want I can suggest you the manifest code here, and then try building it again

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -69,7 +78,7 @@ bench( pkg+':factory', function benchmark( b ) {
 
```
```suggestion
p = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
		p[ i ] = uniform( 0.0, 1.0 );
	}
```

Neerajpathak07 replied:
add `var len`  in the variables list as well

saurabhraghuvanshii replied:
don't understand , we have put the value of y , on next line `y = myquantile(p) ` what should do with this

Neerajpathak07 replied:
uh my bad mistook it as an input variable it should be `p`
Corrected it now

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -69,7 +78,7 @@ bench( pkg+':factory', function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		p = randu();
+		p = uniform( 0.0, 1.0 );
```
```suggestion

```

Neerajpathak07 commented:
@saurabhraghuvanshii The PR is lining up just a few suggested changes and then this should be up and ready!

saurabhraghuvanshii committed:
chore: update benchmark.js manifest.json

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into levy-quantile

saurabhraghuvanshii commented:
@Neerajpathak07 done every thing

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/manifest.json:
```diff
@@ -0,0 +1,85 @@
+          "@stdlib/math/base/special/erfcinv",
+          "@stdlib/constants/float64/ninf",
+          "@stdlib/constants/float64/pinf",
+          "@stdlib/constants/float64/eps"
```
```suggestion
          "@stdlib/math/base/assert/is-nan",
          "@stdlib/math/base/special/erfcinv",
          "@stdlib/constants/float64/eps"
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/manifest.json:
```diff
@@ -0,0 +1,85 @@
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/assert/is-nan",
+          "@stdlib/constants/float64/eps",
+          "@stdlib/constants/float64/ninf",
+          "@stdlib/constants/float64/pinf",
+          "@stdlib/math/base/special/erfcinv"
```
```suggestion
          "@stdlib/math/base/assert/is-nan",
          "@stdlib/math/base/special/erfcinv"
```

Neerajpathak07 replied:
Just these two changes need to be addressed everything else LGTM.

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into levy-quantile

saurabhraghuvanshii committed:
chore: update mainfest.json

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii commented:
done .

saurabhraghuvanshii commented:
@Planeshifter @kgryte sir , can review it , it can be merged as  @Neerajpathak07 review it and I think it is ready for merged .

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
```diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_levy_quantile( p, mu, c )
+
+Returns the quantile function for a Lévy distribution with location parameter `mu`  and scale parameter `c`.
```
```suggestion
Evaluates the [quantile function][quantile-function] for a [Lévy][levy-distribution] distribution with parameters `mu` (location parameter) and `c` (scale parameter).
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/lib/native.js:
```diff
@@ -0,0 +1,76 @@
+
+/**
+* Evaluates the quantile function for a Lévy distribution with location parameter `mu` and scale parameter `c` at a probability `p`.
+*
```
```suggestion
*
* @private
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+		mu = random_uniform( 0.0, 10.0 );
+		c = random_uniform( 0.0, 10.0 );
+		y = stdlib_base_dists_levy_quantile( p, mu, c );
+		printf( "p:%lf, µ: %lf, c: %lf, Q(p;µ,c): %lf\n", p, mu, c, y );
```
```suggestion
		printf( "p: %lf, µ: %lf, c: %lf, Q(p;µ,c): %lf\n", p, mu, c, y );
```

Neerajpathak07 replied:
Might need to take a look for the same in the markdown file.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
````diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+Returns the quantile function for a Lévy distribution with location parameter `mu`  and scale parameter `c`.
+
+```c
+double out = stdlib_base_dists_levy_quantile( 1.1, 0.0, 1.0 );
+// returns NaN
````
```suggestion
double out = stdlib_base_dists_levy_quantile( 0.8, 0.0, 1.0 );
// returns ~15.58
```
In my opinion instead of adding an example which returns `NaN` we can switch to another example which can display an approx value. By doing this we can have a better idea of the function.

Neerajpathak07 commented:
@saurabhraghuvanshii Hey, I have added a few comments based on some recent updates and some which had been overlooked by me for which I apologize. I would really appreciate if you would take a look again.

saurabhraghuvanshii commented:
I am going to do changes as suggested, thanks for reviewing my Pr, it is difficult job to review a Pr , thanks again

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
````diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/levy/quantile.h"
+
+#include <stdlib.h>
+#include <stdio.h>
````
```suggestion
#include "stdlib/stats/base/dists/levy/quantile.h"
#include <stdlib.h>
#include <stdio.h>
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
```diff
@@ -145,6 +145,105 @@ for ( i = 0; i < 10; i++ ) {
+        p = random_uniform( 0.0, 1.0 );
+        mu = random_uniform( 0.0, 10.0 );
+        c = random_uniform( 0.0, 10.0 );
+        y = stdlib_base_dists_levy_quantile( mu, c );
```
```suggestion
        y = stdlib_base_dists_levy_quantile( p, mu, c );
```

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into levy-quantile

saurabhraghuvanshii committed:
chore: updates minor mistakes and fix tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -66,10 +76,14 @@ bench( pkg+':factory', function benchmark( b ) {
 	mu = 10.0;
 	c = 4.0;
 	myquantile = quantile.factory( mu, c );
+	p = new Float64Array( len );
+
+	for ( i = 0; i < len; i++ ) {
```
```suggestion

        len = 100;
	p = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
```

Neerajpathak07 replied:
`len = 100;` should be tab intended

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/benchmark/benchmark.js:
```diff
@@ -66,10 +76,14 @@ bench( pkg+':factory', function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		p = randu();
 		y = myquantile( p );
```
```suggestion
		y = myquantile( p[ i%len ] );
```

saurabhraghuvanshii replied:
These are basic changes how I missed

Neerajpathak07 replied:
Nw you can do them now XD!

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/test/test.quantile.js:
```diff
@@ -102,15 +102,9 @@ tape( 'the function evaluates the quantile function at `p` given positive `mu`',
-		if ( expected[i] !== null ) {
-			if ( y === expected[i] ) {
-				t.equal( y, expected[i], 'p: '+p[i]+', mu:'+mu[i]+', c: '+c[i]+', y: '+y+', expected: '+expected[i] );
-			} else {
-				delta = abs( y - expected[ i ] );
-				tol = 20.0 * EPS * abs( expected[ i ] );
-				t.ok( delta <= tol, 'within tolerance. p: '+p[ i ]+'. mu: '+mu[i]+'. c: '+c[i]+'. y: '+y+'. E: '+expected[ i ]+'. Δ: '+delta+'. tol: '+tol+'.' );
-			}
```
only the non-exact check was meant to be removed same for the `test.quantile.js` file.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/test/test.quantile.js:
```diff
@@ -102,15 +102,9 @@ tape( 'the function evaluates the quantile function at `p` given positive `mu`',
-				tol = 20.0 * EPS * abs( expected[ i ] );
-				t.ok( delta <= tol, 'within tolerance. p: '+p[ i ]+'. mu: '+mu[i]+'. c: '+c[i]+'. y: '+y+'. E: '+expected[ i ]+'. Δ: '+delta+'. tol: '+tol+'.' );
-			}
-		}
```
I believe this should be reverted as, the quantile function involves floating-point arithmetic, which in turn introduces small numerical errors due to precision limitations.

Neerajpathak07 commented:
@saurabhraghuvanshii The PR is in much better shape now!!
Just update your PR tittle to:- 
feat: add C implementation for `stats/base/dists/levy/quantile`

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into levy-quantile

saurabhraghuvanshii committed:
chore: add missing value and revert test.quantile.js

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii commented:
done  .

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/test/test.native.js:
```diff
@@ -0,0 +1,165 @@
+	mu = positiveMean.mu;
+	c = positiveMean.c;
+	for ( i = 0; i < p.length; i++ ) {
+		y = quantile( p[i], mu[i], c[i] );
```
update this as well as the below test cases as done in `test.quantile.js`

Neerajpathak07 replied:
everything else looks good.

saurabhraghuvanshii replied:
same as `test.quantile.js` ?

saurabhraghuvanshii committed:
chore: update test.native.js

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

Neerajpathak07 commented:
LGTM!! Thank you soo much for the changes.
@Planeshifter Kindly take a look at this.

saurabhraghuvanshii commented:
@Neerajpathak07 thanks for reviewing my messing files it is hard work . Thanks again

saurabhraghuvanshii commented:
@kgryte @Planeshifter it is ready for merge and review by @Neerajpathak07 check it

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md:
```diff
@@ -145,6 +145,104 @@ for ( i = 0; i < 10; i++ ) {
+}
+
+int main( void ) {
+    double p;
+    double mu;
```
```suggestion
    double mu;
    double p;
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+}
+
+int main( void ) {
+	double p;
+	double mu;
```
```suggestion
	double mu;
	double p;
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/include/stdlib/stats/base/dists/levy/quantile.h:
```diff
@@ -0,0 +1,38 @@
+extern "C" {
+#endif
+
+/**
+* Returns the quantile for a Lévy distribution with location `mu` and scale `c`.
```
```suggestion
/**
* Evaluates the quantile function for a Lévy distribution with location parameter `mu` and scale parameter `c` at a probability `p`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/src/main.c:
```diff
@@ -0,0 +1,49 @@
+/**
+* Returns the quantile for a Lévy distribution with location `mu` and scale `c`.
+*
+* @param p	  input value
+* @param mu	  location parameter
+* @param c	  scale parameter
+* @return	  evaluated quantile function
```
```suggestion
* @param p   input value
* @param mu  location parameter
* @param c   scale parameter
* @return    evaluated quantile function
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/quantile/src/main.c:
```diff
@@ -0,0 +1,49 @@
+		p < 0.0 ||
+		p > 1.0
+	) {
+		return 0.0 / 0.0; //NaN
```
```suggestion
		return 0.0 / 0.0; // NaN
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: fix indentation

Planeshifter committed:
chore: add missing space

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thank you, @saurabhraghuvanshii, for raising this PR and @Neerajpathak07 for review.

Looks good to me; I will land this shortly after all CI check have passed.
